### DISCUSSION TITLE
0.3.3

### DISCUSSION THREAD

**Discussion by wangxiaoying** [sfu-db/connector-x#628] [Announcement]
## What's Changed
* build(pypi): Constrain the python version by @zen-xu in https://github.com/sfu-db/connector-x/pull/533
* Arrow stream by @wangxiaoying in https://github.com/sfu-db/connector-x/pull/536
* DataFusion 31 and Arrow 46 upgrade by @gruuya in https://github.com/sfu-db/connector-x/pull/540
* Update mssql.md by @albcunha in https://github.com/sfu-db/connector-x/pull/544
* update to `r2d2-oracle=0.6` by @rursprung in https://github.com/sfu-db/connector-x/pull/562
* Infinite Postgres Bug #573  by @AndrewJackson2020 in https://github.com/sfu-db/connector-x/pull/574
* Update SQLite dependencies by @aljazerzen in https://github.com/sfu-db/connector-x/pull/568
* Create dependabot.yml by @tschm in https://github.com/sfu-db/connector-x/pull/576
* Bump actions/cache from 2 to 4 by @dependabot in https://github.com/sfu-db/connector-x/pull/578
* Bump actions/download-artifact from 3 to 4 by @dependabot in https://github.com/sfu-db/connector-x/pull/577
* Bump actions/setup-java from 3 to 4 by @dependabot in https://github.com/sfu-db/connector-x/pull/579
* Bump JamesIves/github-pages-deploy-action from 4.0.0 to 4.5.0 by @dependabot in https://github.com/sfu-db/connector-x/pull/580
* Bump Gr1N/setup-poetry from 8 to 9 by @dependabot in https://github.com/sfu-db/connector-x/pull/581
* Bump actions/upload-artifact from 2 to 4 by @dependabot in https://github.com/sfu-db/connector-x/pull/582
* Bump actions/checkout from 2 to 4 by @dependabot in https://github.com/sfu-db/connector-x/pull/589
* Bump actions/setup-python from 1 to 5 by @dependabot in https://github.com/sfu-db/connector-x/pull/588
* Bump extractions/setup-just from 1 to 2 by @dependabot in https://github.com/sfu-db/connector-x/pull/587
* Update dependencies in cargo.toml using a bot by @tschm in https://github.com/sfu-db/connector-x/pull/583
* build(rust): bump itertools from 0.10.5 to 0.11.0 by @dependabot in https://github.com/sfu-db/connector-x/pull/594
* build(rust): bump openssl from 0.10.63 to 0.10.64 by @dependabot in https://github.com/sfu-db/connector-x/pull/591
* Revert "build(rust): bump openssl from 0.10.63 to 0.10.64" by @wangxiaoying in https://github.com/sfu-db/connector-x/pull/596
* build(rust): bump rust_decimal from 1.33.1 to 1.35.0 by @dependabot in https://github.com/sfu-db/connector-x/pull/597
* Feat/pg semicolon support by @Jordan-M-Young in https://github.com/sfu-db/connector-x/pull/599
* build(rust): bump chrono from 0.4.33 to 0.4.38 by @dependabot in https://github.com/sfu-db/connector-x/pull/602
* add contributor list to readme by @wangxiaoying in https://github.com/sfu-db/connector-x/pull/604
* typing: add stubs for connectorx.so by @zen-xu in https://github.com/sfu-db/connector-x/pull/606
* No need to rely on the backport `importlib_metadata` for version retrieval anymore by @zen-xu in https://github.com/sfu-db/connector-x/pull/607
* typing: support type hint with standard collections by @zen-xu in https://github.com/sfu-db/connector-x/pull/605
* refactor: A more Pythonic way to set environment variables. by @zen-xu in https://github.com/sfu-db/connector-x/pull/608
* Re-write `remove_ending_semicolon` by @surister in https://github.com/sfu-db/connector-x/pull/601
* typing: optimize type annotations by @zen-xu in https://github.com/sfu-db/connector-x/pull/609
* refactor: simplify ternary operator by @zen-xu in https://github.com/sfu-db/connector-x/pull/611
* refactor: Optimize the detection of whether a package is installed. by @zen-xu in https://github.com/sfu-db/connector-x/pull/610
* fix: correct read_sql return type annotation by @zen-xu in https://github.com/sfu-db/connector-x/pull/612
* feat: Supports structured construction of database connection. by @zen-xu in https://github.com/sfu-db/connector-x/pull/613
* build(deps): bump akhilmhdh/contributors-readme-action from 2.3.6 to 2.3.8 by @dependabot in https://github.com/sfu-db/connector-x/pull/618
* build(deps): bump JamesIves/github-pages-deploy-action from 4.5.0 to 4.6.0 by @dependabot in https://github.com/sfu-db/connector-x/pull/617
* implement Trino source by @domnikl in https://github.com/sfu-db/connector-x/pull/585
* [WIP] support python 3.12 by @wangxiaoying in https://github.com/sfu-db/connector-x/pull/615

## New Contributors
* @zen-xu made their first contribution in https://github.com/sfu-db/connector-x/pull/533
* @albcunha made their first contribution in https://github.com/sfu-db/connector-x/pull/544
* @rursprung made their first contribution in https://github.com/sfu-db/connector-x/pull/562
* @AndrewJackson2020 made their first contribution in https://github.com/sfu-db/connector-x/pull/574
* @aljazerzen made their first contribution in https://github.com/sfu-db/connector-x/pull/568
* @tschm made their first contribution in https://github.com/sfu-db/connector-x/pull/576
* @dependabot made their first contribution in https://github.com/sfu-db/connector-x/pull/578
* @surister made their first contribution in https://github.com/sfu-db/connector-x/pull/601
* @domnikl made their first contribution in https://github.com/sfu-db/connector-x/pull/585

**Full Changelog**: https://github.com/sfu-db/connector-x/compare/v0.3.2...v0.3.3

## What's Changed
* build(pypi): Constrain the python version by @zen-xu in https://github.com/sfu-db/connector-x/pull/533
* Arrow stream by @wangxiaoying in https://github.com/sfu-db/connector-x/pull/536
* DataFusion 31 and Arrow 46 upgrade by @gruuya in https://github.com/sfu-db/connector-x/pull/540
* Update mssql.md by @albcunha in https://github.com/sfu-db/connector-x/pull/544
* update to `r2d2-oracle=0.6` by @rursprung in https://github.com/sfu-db/connector-x/pull/562
* Infinite Postgres Bug #573  by @AndrewJackson2020 in https://github.com/sfu-db/connector-x/pull/574
* Update SQLite dependencies by @aljazerzen in https://github.com/sfu-db/connector-x/pull/568
* Create dependabot.yml by @tschm in https://github.com/sfu-db/connector-x/pull/576
* Bump actions/cache from 2 to 4 by @dependabot in https://github.com/sfu-db/connector-x/pull/578
* Bump actions/download-artifact from 3 to 4 by @dependabot in https://github.com/sfu-db/connector-x/pull/577
* Bump actions/setup-java from 3 to 4 by @dependabot in https://github.com/sfu-db/connector-x/pull/579
* Bump JamesIves/github-pages-deploy-action from 4.0.0 to 4.5.0 by @dependabot in https://github.com/sfu-db/connector-x/pull/580
* Bump Gr1N/setup-poetry from 8 to 9 by @dependabot in https://github.com/sfu-db/connector-x/pull/581
* Bump actions/upload-artifact from 2 to 4 by @dependabot in https://github.com/sfu-db/connector-x/pull/582
* Bump actions/checkout from 2 to 4 by @dependabot in https://github.com/sfu-db/connector-x/pull/589
* Bump actions/setup-python from 1 to 5 by @dependabot in https://github.com/sfu-db/connector-x/pull/588
* Bump extractions/setup-just from 1 to 2 by @dependabot in https://github.com/sfu-db/connector-x/pull/587
* Update dependencies in cargo.toml using a bot by @tschm in https://github.com/sfu-db/connector-x/pull/583
* build(rust): bump itertools from 0.10.5 to 0.11.0 by @dependabot in https://github.com/sfu-db/connector-x/pull/594
* build(rust): bump openssl from 0.10.63 to 0.10.64 by @dependabot in https://github.com/sfu-db/connector-x/pull/591
* Revert "build(rust): bump openssl from 0.10.63 to 0.10.64" by @wangxiaoying in https://github.com/sfu-db/connector-x/pull/596
* build(rust): bump rust_decimal from 1.33.1 to 1.35.0 by @dependabot in https://github.com/sfu-db/connector-x/pull/597
* Feat/pg semicolon support by @Jordan-M-Young in https://github.com/sfu-db/connector-x/pull/599
* build(rust): bump chrono from 0.4.33 to 0.4.38 by @dependabot in https://github.com/sfu-db/connector-x/pull/602
* add contributor list to readme by @wangxiaoying in https://github.com/sfu-db/connector-x/pull/604
* typing: add stubs for connectorx.so by @zen-xu in https://github.com/sfu-db/connector-x/pull/606
* No need to rely on the backport `importlib_metadata` for version retrieval anymore by @zen-xu in https://github.com/sfu-db/connector-x/pull/607
* typing: support type hint with standard collections by @zen-xu in https://github.com/sfu-db/connector-x/pull/605
* refactor: A more Pythonic way to set environment variables. by @zen-xu in https://github.com/sfu-db/connector-x/pull/608
* Re-write `remove_ending_semicolon` by @surister in https://github.com/sfu-db/connector-x/pull/601
* typing: optimize type annotations by @zen-xu in https://github.com/sfu-db/connector-x/pull/609
* refactor: simplify ternary operator by @zen-xu in https://github.com/sfu-db/connector-x/pull/611
* refactor: Optimize the detection of whether a package is installed. by @zen-xu in https://github.com/sfu-db/connector-x/pull/610
* fix: correct read_sql return type annotation by @zen-xu in https://github.com/sfu-db/connector-x/pull/612
* feat: Supports structured construction of database connection. by @zen-xu in https://github.com/sfu-db/connector-x/pull/613
* build(deps): bump akhilmhdh/contributors-readme-action from 2.3.6 to 2.3.8 by @dependabot in https://github.com/sfu-db/connector-x/pull/618
* build(deps): bump JamesIves/github-pages-deploy-action from 4.5.0 to 4.6.0 by @dependabot in https://github.com/sfu-db/connector-x/pull/617
* implement Trino source by @domnikl in https://github.com/sfu-db/connector-x/pull/585
* support python 3.12 by @wangxiaoying in https://github.com/sfu-db/connector-x/pull/615

## New Contributors
* @zen-xu made their first contribution in https://github.com/sfu-db/connector-x/pull/533
* @albcunha made their first contribution in https://github.com/sfu-db/connector-x/pull/544
* @rursprung made their first contribution in https://github.com/sfu-db/connector-x/pull/562
* @AndrewJackson2020 made their first contribution in https://github.com/sfu-db/connector-x/pull/574
* @aljazerzen made their first contribution in https://github.com/sfu-db/connector-x/pull/568
* @tschm made their first contribution in https://github.com/sfu-db/connector-x/pull/576
* @dependabot made their first contribution in https://github.com/sfu-db/connector-x/pull/578
* @surister made their first contribution in https://github.com/sfu-db/connector-x/pull/601
* @domnikl made their first contribution in https://github.com/sfu-db/connector-x/pull/585

**Full Changelog**: https://github.com/sfu-db/connector-x/compare/v0.3.2...v0.3.3

<hr /><em>This discussion was created from the release <a href='https://github.com/sfu-db/connector-x/releases/tag/v0.3.3'>0.3.3</a>.</em>

**Comment by aleewen** [comment]
Thank you so much for this!
Could we please get the source distribution up on PyPI? It's been causing us a lot of problems.
